src/bidiMapper/domains/network/networkRequest.ts

```diff
@@ -29,54 +28,76 @@ import {Network} from '../../../protocol/protocol.js';
 
 export class NetworkRequest {
   static #unknown = 'UNKNOWN';
-  requestId: string;
+
+  /**
+   * Each network request has an associated request id, which is a string
+   * uniquely identifying that request.
+   *
+   * The identifier for a request resulting from a redirect matches that of the
+   * request that initiated it.
+   */
+  requestId: Network.Request;
+
   #servedFromCache = false;
+  #redirectCount: number;
+
   #eventManager: IEventManager;
-  #requestWillBeSentEvent: Protocol.Network.RequestWillBeSentEvent | undefined;
-  #requestWillBeSentExtraInfoEvent:
-    | Protocol.Network.RequestWillBeSentExtraInfoEvent
-    | undefined;
-  #responseReceivedEvent: Protocol.Network.ResponseReceivedEvent | undefined;
 
-  #responseReceivedExtraInfoEvent:
-    | Protocol.Network.ResponseReceivedExtraInfoEvent
-    | undefined;
+  #requestWillBeSentEvent?: Protocol.Network.RequestWillBeSentEvent;
+  #requestWillBeSentExtraInfoEvent?: Protocol.Network.RequestWillBeSentExtraInfoEvent;
+  #responseReceivedEvent?: Protocol.Network.ResponseReceivedEvent;
+  #responseReceivedExtraInfoEvent?: Protocol.Network.ResponseReceivedExtraInfoEvent;
 
-  #beforeRequestSentDeferred: Deferred<void>;
+  #requestWillBeSentDeferred: Deferred<void>;
```

Comment:
How I viewed it is that this is that Deferred are for the output BiDi commands, this one being `network.beforeRequestSent`

Reply:
OK, reverted